December improvements (#19)
cachix/cachix-action: README.md

Build software only once using [Nix](https://nixos.org/nix/) with the help of [Cachix](https://cachix.org).

This action will configure Cachix and invoke `nix-build`.

## Why do I need this

Because you'd like for your CI to be fast. Let me explain.

Directory-based caching on a typical CI doesn't work well for Nix.

`/nix/store` is a global storage of everything Nix operates on. These are
your sources, patches, tarballs, packages, configuration.

> A directory-based cache requires that downloading a whole store, including the irrelevant parts. cachix-action will only fetch what's needed by configuring a Nix binary cache.

Reviewer: ```suggestion
A directory-based cache requires downloading a whole store, including the irrelevant parts. `cachix-action` will only fetch what's needed by configuring a Nix binary cache.
```

Build software only once using [Nix](https://nixos.org/nix/) with the help of [Cachix](https://cachix.org).

This action will configure Cachix and invoke `nix-build`.

## Why do I need this

Because you'd like for your CI to be fast. Let me explain.

> Caching on a typical CI doesn't work in favor of Nix.

Reviewer: ```suggestion
Directory-based caching on a typical CI doesn't work for Nix.
```

Author: That's a strong statement, it should then say "work well".

`/nix/store` is a global storage of everything Nix operates on. These are
your sources, patches, tarballs, packages, configuration.

> Caching `/nix/store` is time consuming as Nix only appends store paths to it.

Reviewer: > Caching `/nix/store` is time consuming as Nix only appends store paths to it.

This sentence didn't help me at all.

Here's my perspective as a comparison. Needs some polish too.

A directory-based cache requires that downloading a whole store, including the irrelevant parts. `cachix-action` will only fetch what's needed by configuring a Nix binary cache.

When the build is done, `cachix-action` only has to upload the new store paths, rather than syncing the whole store.

Purging paths from a directory-based cache is not feasible because it'd have to be aware of all branches and their respective contents somehow.

        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
```

Alternatively, you can use this action to only configure cachix for substitution:

```yaml
...

>     - uses: cachix/cachix-action@v2

Reviewer 2: you probably mean `v3`?

Author: Will bump that once release is ready :)
